core/chaincode/platforms/golang/hash.go

```diff
@@ -138,7 +135,11 @@ func getCodeFromHTTP(path string) (codegopath string, err error) {
 	//     . more secure
 	//     . as we are not downloading OBC, private, password-protected OBC repo's become non-issue
 
-	env[gopathenvIndex] = "GOPATH=" + codegopath + ":" + origgopath
+	os.Setenv("GOPATH", codegopath+":"+origgopath)
+	// Get a copy of that new env for the go get command
+	env := os.Environ()
+	// and reset GOPATH to its original value
+	os.Setenv("GOPATH", origgopath)
```

Comment:
Ok, I think I found a happy medium. Reusing the old way of setting the environment for the subcommand in a threadsafe manner while still ending up with a simpler piece of code.
And with the new commit the ci tests kicked in so that will address any possible problems with the addition of a comment I previously made with ci skip.